tests/summ_ai_api/utils.py

```diff
@@ -0,0 +1,94 @@
+from asgiref.sync import sync_to_async
+import aiohttp
+
+from integreat_cms.cms.models import Page, Region
+
+
+def make_mock_summ_ai_server(response_data, status=200):
+    """
+    Creates a mock of SummAiServer-(Response)
+
+    :param response_data: data that mocked server returns
+    :type response_data: dict
+
+    :param status: reponse_data that mocked server returns (default = 200)
+    :type status: int
+
+    :return: mocked SummAiServer
+    :rtype: aiohttp.web.Response
+    """
+
+    async def fake_summ_ai_server(request):
+        """
+        Create fake responses which simulate the SUMM.AI API server
+
+        :param request: The request
+        :type request: aiohttp.web.Request
+
+        :return: The response
+        :rtype: aiohttp.web.Response
+        """
+        return aiohttp.web.json_response(data=response_data, status=status)
+
+    return fake_summ_ai_server
+
```

Comment:
```suggestion


def make_mock_summ_ai_server_rate_limited():
    """
    Creates a mock of SUMM.AI server with rate limiting

    :return: Mocked SUMM.AI server with rate limiting
    :rtype: aiohttp.web.Application
    """
    app = aiohttp.web.Application()
    app["attempt"] = 0

    async def fake_summ_ai_server_rate_limited(request):
        request.app["attempt"] += 1
        if request.app["attempt"] < 3:
            return aiohttp.web.json_response(
                data={"error": "rate limit exceeded"}, status=429
            )
        return aiohttp.web.json_response(
            data={
                "translated_text": "Hier ist Ihre Leichte Sprache Übersetzung",
                "jobid": "9999",
            },
            status=200,
        )

    app.router.add_post("/", fake_summ_ai_server_rate_limited)
    return app


```

Since the queue is still not working as expected despite all tests succeeding, I fear we can not avoid adding an integration test for this any longer. I suggest e.g. this fake server which combines the ideas of `test_worker()` and `make_mock_summ_ai_server()`.